node/node.go

```diff
@@ -703,6 +703,10 @@ func (n *nodeImpl) OnStart() error {
 			state, err := n.stateSyncReactor.Sync(context.TODO())
 			if err != nil {
 				n.Logger.Error("state sync failed", "err", err)
```

Comment:
Consider maybe also logging that we are shutting down the node?

```suggestion
				n.Logger.Error("state sync failed; shutting down this node", "err", err)
```